aslam_cv_cameras/src/ncamera.cc

```diff
@@ -3,24 +3,28 @@
 
 #include <glog/logging.h>
 
-#include <aslam/cameras/camera.h>
+#include <aslam/cameras/camera-factory.h>
 #include <aslam/cameras/camera-pinhole.h>
+#include <aslam/cameras/camera-unified-projection.h>
+#include <aslam/cameras/camera.h>
 #include <aslam/cameras/distortion-radtan.h>
 #include <aslam/cameras/ncamera.h>
-#include <aslam/cameras/yaml/ncamera-yaml-serialization.h>
+#include <aslam/cameras/random-camera-generator.h>
 #include <aslam/common/pose-types.h>
 #include <aslam/common/predicates.h>
 #include <aslam/common/unique-id.h>
+#include <aslam/common/yaml-serialization.h>
 
 namespace aslam {
 
-NCamera::NCamera(const NCameraId& id, const TransformationVector& T_C_B,
-                 const std::vector<Camera::Ptr>& cameras, const std::string& label)
-    : id_(id),
-      T_C_B_(T_C_B),
-      cameras_(cameras),
-      label_(label) {
+NCamera::NCamera() {}
+
+NCamera::NCamera(
+    const NCameraId& id, const TransformationVector& T_C_B,
+    const std::vector<Camera::Ptr>& cameras, const std::string& description)
+    : Sensor(id), T_C_B_(T_C_B), cameras_(cameras) {
   CHECK(id.isValid());
+  description_ = description;
```

Comment:
It's a bit trickier since description is a member of the base Sensor class and cannot therefore be in the initializer list of NCamera. I added a new constructor and now should be nicer if you want to check the solution looks good to you?

Reply:
ah i see. yeah this looks good to me. Depending on the usage we could also think about making this new constructor protected then. what do you think?

Reply:
I think it's ok since for now all the constructors in sensor are valid options and anyways a sensor class can't be constructed on it's own due the the purely virtual functions.